netdeploy/network.go

```diff
@@ -57,25 +58,37 @@ type Network struct {
 
 // CreateNetworkFromTemplate uses the specified template to deploy a new private network
 // under the specified root directory.
-func CreateNetworkFromTemplate(name, rootDir, templateFile, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
+func CreateNetworkFromTemplate(name, rootDir string, templateReader io.Reader, binDir string, importKeys bool, nodeExitCallback nodecontrol.AlgodExitErrorCallback, consensus config.ConsensusProtocols, overrideDevMode bool) (Network, error) {
 	n := Network{
 		rootDir:          rootDir,
 		nodeExitCallback: nodeExitCallback,
 	}
 	n.cfg.Name = name
-	n.cfg.TemplateFile = templateFile
 
-	template, err := loadTemplate(templateFile)
+	var err error
+	var template NetworkTemplate
+
+	err = loadTemplateFromReader(templateReader, &template)
+
 	if err == nil {
 		if overrideDevMode {
 			template.Genesis.DevMode = true
+			if template.Genesis.NetworkName == "" {
```

Comment:
I'm going back and forth on this approach. What do you think about these options:
1. Put "Default Network Template" in the default template.
2. Allow the command to fail, the user would have to provide the `--network` option to name their network.

Reply:
I don't have a strong preference.

Reply:
> 1. Put "Default Network Template" in the default template.
> 2. Leave the network name blank and allow the command to fail. The user would be prompted to provide the --network option to name their network.

I like this--it's maybe less convenient, but I like fewer magic values like "devmodenet".

Reply:
Which option do you prefer? I lean a little towards 1 so that the command works with fewer arguments.

Reply:
Put "Default Network Template" in the default template and allow everything else to fail.